rust/stackable-cockpit/src/platform/stacklet/mod.rs

```diff
@@ -164,42 +168,15 @@ async fn list_stackable_stacklets(
     Ok(stacklets)
 }
 
-fn build_products_gvk_list<'a>(product_names: &[&'a str]) -> IndexMap<&'a str, GroupVersionKind> {
-    let mut map = IndexMap::new();
-
-    for product_name in product_names {
-        // Note(techassi): Why? Just why? Can we please make this consistent?
-        // Note(sbernauer): I think it's legit that SparkHistoryServer and SparkConnectServer are in
-        // the api group spark.stackable.tech. All of this will probably be rewritten any as soon as
-        // we have versions different than v1alpha1.
-        if *product_name == "spark-history" {
-            map.insert(*product_name, GroupVersionKind {
-                group: "spark.stackable.tech".into(),
-                version: "v1alpha1".into(),
-                kind: "SparkHistoryServer".into(),
-            });
-        } else if *product_name == "spark-connect" {
-            map.insert(*product_name, GroupVersionKind {
-                group: "spark.stackable.tech".into(),
-                version: "v1alpha1".into(),
-                kind: "SparkConnectServer".into(),
-            });
-        } else {
-            map.insert(*product_name, gvk_from_product_name(product_name));
-        }
-    }
-
-    map
-}
-
-// FIXME: Support SparkApplication and SparkConnectServer
-fn gvk_from_product_name(product_name: &str) -> GroupVersionKind {
-    GroupVersionKind {
-        group: format!("{product_name}.stackable.tech"),
-        version: "v1alpha1".into(),
-        kind: format!(
-            "{product_name}Cluster",
-            product_name = product_name.capitalize()
-        ),
-    }
+fn gvk_from_product_name(product_name: &str) -> Result<GroupVersionKind, Error> {
+    let (_, group, version, kind) = PRODUCTS
+        .iter()
+        .find(|(other_product_name, _, _, _)| product_name == *other_product_name)
+        .context(GetProductSnafu { product_name })?;
+
+    Ok(GroupVersionKind {
+        group: group.to_string(),
+        version: version.to_string(),
+        kind: kind.to_string(),
+    })
```

Comment:
suggestion: Use `to_owned` here instead.


```suggestion
    Ok(GroupVersionKind {
        group: group.to_owned(),
        version: version.to_owned(),
        kind: kind.to_owned(),
    })
```

Reply:
Resolved, see https://github.com/stackabletech/stackable-cockpit/pull/400#discussion_r2209900374.